sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -582,6 +582,15 @@ object SQLConf {
     .intConf
     .createWithDefault(2667)
 
+  val CODEGEN_MAX_CHARS_PER_FUNCTION = buildConf("spark.sql.codegen.maxCharactersPerFunction")
```

Comment:
If the length of source code does not mean the number of lines of the source code, you are right.
This is because we check the sum of [`String.length`](https://github.com/apache/spark/pull/18966/files#diff-8bcc5aea39c73d4bf38aef6f6951d42cR785). Here are two examples.
```abc``` -> 3
```
ab
c
```
-> 4